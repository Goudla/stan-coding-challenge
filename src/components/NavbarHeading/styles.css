.navbar-heading h1 {
  color: #fff;
  font-size: 33px;
  font-weight: 600;
  margin: 0;
  padding: 12px 0;
  text-shadow: 0 1px 3px rgba(0,0,0,0.3);
}

.navbar-heading a {
  text-decoration: none;
}
