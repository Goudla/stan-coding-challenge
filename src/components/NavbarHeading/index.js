import * as React from 'react';
import { Link } from "react-router-dom";

// Styles
import './styles.css';

type Props = {
  children: React.Node
};

class NavbarHeading extends React.PureComponent<Props> {
  render() {
    const { children } = this.props;

    return (
      <span className="navbar-heading">
        <Link to="/">
          <h1>
              {children}
          </h1>
        </Link>
      </span>
    );
  }
}

export default NavbarHeading;
