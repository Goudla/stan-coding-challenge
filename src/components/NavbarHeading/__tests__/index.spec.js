import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

// Scenes
import NavbarHeading from '../';

describe('<NavbarHeading />', () => {
  it('Should render', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<NavbarHeading>Test</NavbarHeading>);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
