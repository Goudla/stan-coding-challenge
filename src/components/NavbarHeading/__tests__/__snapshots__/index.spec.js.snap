// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavbarHeading /> Should render 1`] = `
<span
  className="navbar-heading"
>
  <Link
    to="/"
  >
    <h1>
      Test
    </h1>
  </Link>
</span>
`;
